src/coreclr/src/zap/zapinfo.cpp

```diff
@@ -502,50 +502,61 @@ void ZapInfo::CompileMethod()
     ULONG cCode;
 
 #ifdef ALLOW_SXS_JIT_NGEN
+    bool expectedAltJitFailure = false;
```

Comment:
Why do we need this fallback after other changes in this PR?

Reply:
Whenever I generate AltJit jit-diffs, we always get two copies of every method.
This fixes that issue.  I could move this into it's own checkin.

Reply:
Yes, please. I have checked that I see that behavior (crossgen only), but I will need more time to understand your fix, so it will be nice not to block the PR because of this change.